src/microcalibrate/calibration.py

```diff
@@ -463,3 +487,916 @@ def summary(
+
+            Args:
+                holdout_set: Dictionary with 'names' and 'indices' of holdout targets
+                hyperparameters: Dictionary with l0_lambda, init_mean, temperature
+                epochs_per_trial: Number of epochs to run
+                objectives_balance: Weights for different objectives
+
+            Returns:
+                Dictionary with evaluation metrics and holdout target names
+            """
+            # Store original parameters
+            original_params = {
+                "l0_lambda": self.l0_lambda,
+                "init_mean": self.init_mean,
+                "temperature": self.temperature,
+                "regularize_with_l0": self.regularize_with_l0,
+                "epochs": self.epochs,
+            }
+
+            try:
+                # Update parameters for this evaluation
+                self.l0_lambda = hyperparameters["l0_lambda"]
+                self.init_mean = hyperparameters["init_mean"]
+                self.temperature = hyperparameters["temperature"]
+                self.regularize_with_l0 = True
+                self.epochs = epochs_per_trial
+
+                # Set up calibration with this holdout set
+                self.excluded_targets = holdout_set["names"]
+                self.exclude_targets()
+
+                # Run calibration
+                self.calibrate()
+                sparse_weights = self.sparse_weights
+
+                # Get estimates for all targets
+                weights_tensor = torch.tensor(
+                    sparse_weights, dtype=torch.float32, device=self.device
+                )
+
+                if self.original_estimate_matrix is not None:
+                    original_matrix_tensor = torch.tensor(
+                        self.original_estimate_matrix.values,
+                        dtype=torch.float32,
+                        device=self.device,
+                    )
+                    all_estimates = (
+                        (weights_tensor @ original_matrix_tensor).cpu().numpy()
+                    )
+                else:
+                    all_estimates = (
+                        self.original_estimate_function(weights_tensor)
+                        .cpu()
+                        .numpy()
+                    )
+
+                # Split into train/validation
+                n_targets = len(self.original_target_names)
+                val_indices = holdout_set["indices"]
+                train_indices = [
+                    i for i in range(n_targets) if i not in val_indices
+                ]
+
+                val_estimates = all_estimates[val_indices]
+                val_targets = self.original_targets[val_indices]
+                train_estimates = all_estimates[train_indices]
+                train_targets = self.original_targets[train_indices]
+
+                # Calculate metrics
+                from .utils.metrics import loss, pct_close
```

Comment:
If you wouldn't mind, spelling out `from microcalibrate.utils.metrics import loss, pct_close` makes it easier for me to run interactively. Also, the AIs are usually telling me to declare my imports at the top. Is it really doing much for us to delay this import until now (I can see it for `from policyengine_us import Microsimulation`, because it takes like 7 seconds).